yamux/src/stream.rs

```diff
@@ -449,18 +449,20 @@ impl AsyncWrite for StreamHandle {
 
 impl Drop for StreamHandle {
     fn drop(&mut self) {
-        if !self.unbound_event_sender.is_closed()
-            && self.state != StreamState::Closed
-            && self.state != StreamState::LocalClosing
-        {
-            let mut flags = self.get_flags();
-            flags.add(Flag::Rst);
-            let frame = Frame::new_window_update(flags, self.id, 0);
-            let rst_event = StreamEvent::Frame(frame);
+        if !self.unbound_event_sender.is_closed() && self.state != StreamState::Closed {
             let event = StreamEvent::Closed(self.id);
-            // Always successful unless the session is dropped
-            let _ignore = self.unbound_event_sender.unbounded_send(rst_event);
-            let _ignore = self.unbound_event_sender.unbounded_send(event);
+            if self.state == StreamState::LocalClosing {
+                let _ignore = self.unbound_event_sender.unbounded_send(event);
```

Comment:
Some comments can be added here to explain why `rst_event` is not required when `LocalClosing`.